src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -510,6 +510,14 @@ void ResolveContext::OnNewInterface(uint32_t interfaceId, const char * fullname,
     // resolving.
     interface.fullyQualifiedDomainName = hostnameWithDomain;
 
+    // DNSServiceResolve caches old entries with the same interface ID. This means that when the broadcast with the
+    // same interface has been updated then it will not be reflected in the list of interfaces. In this case, remove
+    // the existing entry in the map since the resolve seems to store it in sequential order and the last entry is
+    // the most up to date.
+    if ((interfaces.find(interfaceId)) != interfaces.end())
+    {
+        interfaces.erase(interfaceId);
```

Comment:
What tells us that this interfaceId is no longer valid and is ok to erase?

Reply:
It's being replaced, only one interface definition can be used so we're opting for the one that comes later in the sequence here rather than just straight erasing it. Adding to a map where the item already exists will result in nothing happening.

Reply:
@bzbarsky-apple / @vivien-apple Can either of you confirm the behavior i'm seeing? The way i'm testing is by having a server broadcasting the `_matterd._udp` broadcast then changing the broadcast name. In the logs I'm seeing `DNSServiceResolve` returning 2 entries, one from the old brodcast name and one from the new one.

My assumption here is that the function is caching entries internally and sending back old data then new data. It only seems to reset when i kill the app and re-open it (i guess this clears out the cache?).

Reply:
How are you "changing the broadcast name"?

I am happy to try to reproduce, but I would need clear step-by-step instructions.

> My assumption here is that the function is caching entries internally 

This is a good assumption.

> and sending back old data then new data

This seems like a bad assumption.

> It only seems to reset when i kill the app and re-open it

Which app?  The cache is OS-wide, in general....